spec/mongo/index/view_spec.rb

```diff
@@ -196,7 +196,70 @@
         expect(view.get('random_name')).to_not be_nil
       end
     end
-  end
+
+    context 'when providing an invalid partial index filter' do
+
+      let(:spec) do
+        { random:1 }
+      end
+
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: 5)
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {"x": {"$asda": 3}})
+        }.to raise_error(Mongo::Error::OperationFailure)
+      end
+      it 'raises an exception' do
+        expect {
+          view.create_one(spec, partial_filter_expression: {
+                                  "$and": [{"$and": [{"x": {"$lt": 2}},
+                                                     {"x": {"$gt": 0}}]},
```

Comment:
Check out the failures on Travis - this syntax isn't valid in older versions of Ruby.
For example: https://travis-ci.org/mongodb/mongo-ruby-driver/jobs/83954684